code/modules/surgery/organs/augments_internal.dm

```diff
@@ -137,6 +137,23 @@
 /obj/item/organ/cyberimp/brain/anti_stun/proc/reboot()
 	broken_cyber_organ = FALSE
 
+/obj/item/organ/cyberimp/brain/metal_plate
+	name = "plate of surgical metal"
+	desc = "A small plate of stainless steel, designed to replace missing skull pieces and to calm down weirdos with demons in their heads."
+	slot = ORGAN_SLOT_METAL_HEAD_PLATE
+
+/obj/item/organ/cyberimp/brain/metal_plate/Initialize()
+	. = ..()
+	AddComponent(/datum/component/anti_magic, FALSE, FALSE, TRUE, null, 12, TRUE, null, CALLBACK(src, .proc/use_up))
+
+/obj/item/organ/cyberimp/brain/metal_plate/Remove(mob/living/carbon/M, special = FALSE)
+	. = ..()
+	to_chat(M, "<span class='notice'>Your skull suddenly feels lighter!</span>")
+
+/obj/item/organ/cyberimp/brain/metal_plate/proc/use_up()
+	qdel(GetComponent(/datum/component/anti_magic))
```

Comment:
The anti_magic component should be getting deleted one way or the other if it expires. That should be handled from inside the component